tests/h/views/api_groups_test.py

```diff
@@ -100,12 +100,80 @@ def open_groups(self, factories):
         return [factories.OpenGroup(), factories.OpenGroup()]
 
     @pytest.fixture
-    def anonymous_request(self, pyramid_request):
-        pyramid_request.user = None
+    def authenticated_request(self, pyramid_request, factories):
+        pyramid_request.user = factories.User()
         return pyramid_request
 
+
+@pytest.mark.usefixtures('CreateGroupAPISchema',
+                         'group_service',
+                         'GroupContext',
+                         'GroupJSONPresenter')
+class TestCreateGroup(object):
+
+    def test_it_inits_group_create_schema(self, pyramid_request, CreateGroupAPISchema):  # noqa: N803
+        views.create(pyramid_request)
+
+        CreateGroupAPISchema.assert_called_once()
+
+    # @TODO Move this test once _json_payload() has been moved to a reusable util module
+    def test_it_raises_if_json_parsing_fails(self, pyramid_request):
+        """It raises PayloadError if parsing of the request body fails."""
+        # Make accessing the request.json_body property raise ValueError.
+        type(pyramid_request).json_body = {}
+        with mock.patch.object(type(pyramid_request),
+                               'json_body',
+                               new_callable=mock.PropertyMock) as json_body:
+            json_body.side_effect = ValueError()
+            with pytest.raises(views.PayloadError):
+                views.create(pyramid_request)
+
+    def test_it_returns_400_if_user_has_non_default_authority(self, pyramid_request, factories):
+        pyramid_request.user = factories.User(authority='other.com')
+
+        with pytest.raises(HTTPBadRequest):
+            views.create(pyramid_request)
+
+    def test_it_proxies_to_group_create_service(self, pyramid_request, group_service):
+        views.create(pyramid_request)
+
+        group_service.create_private_group.assert_called_once()
```

Comment:
This test is probably unnecessary given the one below it?

Reply:
Ah, yes. Good point. This happened because I was truly TDD-ing and things were growing as I went :).